build.gradle

```diff
@@ -46,9 +46,14 @@ subprojects {
     task delombok(type: io.franzbecker.gradle.lombok.task.DelombokTask) {
         def outputDir = file("$buildDir/delombok")
         outputs.dir(outputDir)
+        outputs.cacheIf {
+            true
+        }
         for (srcDir in project.sourceSets.main.java.srcDirs) {
+            // TODO: named input and relative path for `srcDir` to make task more cacheable
             inputs.dir(srcDir)
-            args(srcDir, "-d", outputDir)
+            // TODO: `outputDir` as relative path to make task more cacheable 
+            args(srcDir, "-d", outputDir, "-f", "generateDelombokComment:skip")
```

Comment:
`generateDelombokComment:skip`
^ what's the impact for users when viewing source?

Is this necessary for cacheability?

Reply:
What this did was adding a comment to the top of each generated file (something int :
```
// Generated by delombok at Tue Nov 09 17:39:03 CET 2021
```

From the Lombok docs:
````
"Either 'generate' or 'skip'. generate means: Any file modified by delombok will have a comment stating this at the top. Default: 'generate'"
```

This makes it impossible to cache tasks that depend on `delombok` output (in our case, the `javadoc` task).

It should AFAIK have no impact on the users to disable this.

Reply:
Aha, so the published source JAR is not affected?

Reply:
Affected in this way that it does not contain the leading comment:
```
// Generated by delombok at Tue Nov 09 17:39:03 CET 2021
```

I would not think that any tooling would care about this preamble?

Reply:
I have a horrible feeling that IDEA has a special case for this and toggles our old nemesis, the red banner. After `publishToMavenLocal` and using the published vs PR versions in our repro project:

`1.16.2`:
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/132175/145374468-1865c1fb-f689-46c4-89a1-ae69a432b4c5.png">

`unspecified` (branch build):
<img width="933" alt="image" src="https://user-images.githubusercontent.com/132175/145374512-3b9b74af-e1e9-4c12-9b75-9c8ba7de29ed.png">

Reply:
Actually - ignore this. Something else must be having an effect, because this comment is not present in the published `1.16.2` sources JAR either.